doc/specs/#976 - VT52 escape sequences.md

```diff
@@ -197,6 +197,18 @@ z               |₆              |U+2086         |Subscript 6
 }               |₉              |U+2089         |Subscript 9
 \~              |¶              |U+00B6         |Paragraph
 
+### Testing
+
+A simple unit test will need to be added to the `AdapterTest` class, to confirm that calls to toggle between the ANSI and VT52 modes in the `AdaptDispatch` class are correctly forwarded to the corresponding `PrivateXXX` handler in the `ConGetSet` interface.
+
+The majority of the testing would be handled in the `StateMachineExternalTest` class though. These tests would confirm that the various VT52 sequences trigger the expected methods in the `ITermDispatch` interface when VT52 Mode is enabled, and also that they don't do anything when in ANSI mode.
+
+There shouldn't really be any need for additional tests in the `ScreenBufferTests` class, since we're relying on existing VT100 functionality which should already be tested there.
+ 
+For fuzzing support, we'll need to add the DECANM option to the `GeneratePrivateModeParamToken` method in the `VTCommandFuzzer` class, and also probably add two additional token generator methods - one specifically for the _Direct Cursor Address_ command, which requires parameters, and another to handle the remaining parameterless commands.
+
+In terms of manual testing, it can be useful to run the _Test of VT52 mode_ option in Vttest, and confirm that everything looks correct there. It's also worth going through some of the options in the The _Test of keyboard_ section, since those tests aren't only intended for the later VT models - they do cover the VT52 keyboard as well.

```

Comment:
I wonder if we could make a future work item to try to leverage `Vttest` in our automated testing...

I'm OK with the manual directive for now, but this looks like an opportunity for future automation.

Reply:
Issue #1746 might be a good place for further discussion on this topic. I'm not that keen on the Javascript service concept that was being proposed there, but the general idea of being able to reuse third-party test scripts sounded promising.

For example, if you look at the [libvterm project](https://github.com/neovim/libvterm/tree/master/t) that was linked, they have a number of test scripts that essentially replicate the vttest screens. If there was a way for our test framework to consume scripts like that, I think that would probably be easier to work with than automating vttest itself.